packages/apollo-gateway/src/datasources/__tests__/RemoteGraphQLDataSource.test.ts

```diff
@@ -125,6 +173,7 @@ describe('error handling', () => {
 
     const result = DataSource.process({
       request: { query: '{ me { name } }' },
+      context: {},
```

Comment:
Should we continue to test the existing case where `context` was not present?

Reply:
The addition of these empty objects is in response to TS errors that exist on master. Including these changes within this PR may be incorrect, but I do think they should be here.

The outstanding issue is that context is a _required_ field according to the types. What I'm curious to know is, since we previously guaranteed the _runtime_ validity of a missing `context` entry in our tests, should we continue to do so? I'm tempted to say no, as this seems like a pandora's box of runtime null/undefined expectations. The core idea being: just because it works, doesn't mean it's expected behavior.

I'd like to hear your thoughts, and happy to make these changes in a separate PR to prevent conflating the issue!

Reply:
> I'm tempted to say no, as this seems like a pandora's box of runtime null/undefined expectations.

Not everyone is using TypeScript though, so I do believe there is value in testing exactly this.

For speed, I'm okay with leaving them here, though if this PR is squashed, it will be hard to identify why these were changed later. 😉

Reply:
I separated this commit specifically for the sake of revertibility, so consider it done :)

I agree, and feel that your consideration here errs on the side of caution, so I'll revert for now and reconsider at a later time. I think there's some philosophical discussion to be had w.r.t. un/expected behavior, documentation, and types. I definitely understand the "not everyone uses TypeScript" argument, but there may be some nuance here for me to understand better.